src/ray/raylet/node_manager.cc

```diff
@@ -766,7 +766,10 @@ void NodeManager::HandleWorkerBlocked(std::shared_ptr<Worker> worker) {
 }
 
 void NodeManager::HandleWorkerUnblocked(std::shared_ptr<Worker> worker) {
-  RAY_CHECK(worker->IsBlocked());
+  RAY_CHECK(worker);
+  if (!worker->IsBlocked()) {
```

Comment:
Why has this changed? Shouldn't this only be called when the worker is blocked?

Reply:
In legacy ray, a worker always gets marked as blocked if it sends a `ReconstructObjects` message, so once it sends the `NotifyUnblocked` message, it's guaranteed to be blocked. The thing that I changed here compared to legacy ray is that the worker only gets marked as blocked if there is an object requested in the `ReconstructObjects` message that is not local. That means that sometimes, an unblocked worker will send the `NotifyUnblocked` message.